### PR TITLE
DRIVERS-1293: Updated auth spec to use hello and legacy hello

### PR DESCRIPTION
**Pull request by JamesKovacs**

### CODE REVIEW

File: source/auth/auth.rst [hunk 1]
```diff
@@ -1129,9 +1130,9 @@ For a non-existent username, verify that not specifying a mechanism when
 connecting fails with the same error type that would occur with a correct
 username but incorrect password or mechanism.  (Because negotiation with a
 non-existent user name at one point during server development caused an
```
Comment (emadum) [inline source/auth/auth.rst:1132]:
```suggestion
non-existent user name at one point during server development caused a
```

**Review by emadum — APPROVED** [review]
LGTM

File: source/auth/auth.rst [hunk 2]
```diff
@@ -231,13 +232,13 @@ When authenticating each credential, if the authentication mechanism is not
...
-  later, drivers MUST send a new ``isMaster`` negotiation command for the
-  credential to determine the default authentication mechanism.
+  later, drivers MUST send a new ``hello`` or legacy hello negotiation command
+  for the credential to determine the default authentication mechanism.
 
 - Otherwise, when the server version is earlier than 4.0, the driver MUST
   select a default authentication mechanism for the credential following the
   instructions for when the ``saslSupportedMechs`` field is not present in
-  an ``isMaster`` response.
+  an legacy hello response.
```
Comment (bazile-clyde) [inline source/auth/auth.rst:241]:
```suggestion
  a legacy hello response.
```

**Review by bazile-clyde — APPROVED** [review]
LGTM with one nit.

File: source/auth/auth.rst (hunk 1 above)
Comment (JamesKovacs) [reply source/auth/auth.rst:1132]: Fixed

File: source/auth/auth.rst (hunk 2 above)
Comment (JamesKovacs) [reply source/auth/auth.rst:241]: Fixed

**Pull request merged**